tsdb/tx.go

```diff
@@ -248,15 +254,27 @@ func (l *LocalMapper) Open() error {
 	l.txn = txn
 
 	// create a bolt cursor for each unique series id
-	l.cursors = make([]*bolt.Cursor, len(l.seriesKeys))
+	l.cursors = make([]*shardCursor, len(l.seriesKeys))
 
 	for i, key := range l.seriesKeys {
+		// Retrieve key bucket.
 		b := l.txn.Bucket([]byte(key))
-		if b == nil {
+
+		// Ignore if there is no bucket or points in the cache.
+		if b == nil && len(l.shard.cache[key]) == 0 {
 			continue
 		}
 
-		l.cursors[i] = b.Cursor()
+		// Retrieve a copy of the in-cache points for the key.
+		cache := make([][]byte, len(l.shard.cache[key]))
+		copy(cache, l.shard.cache[key])
```

Comment:
Flush time should probably be configurable. For users that have metrics collecting once a minute, they'll probably want this to be higher like maybe `20m`.

It's not safe to assume we're write heavy. For example you have monitoring systems that are doing regular queries against recent data, user dashboards pulling recent data, and continuous queries hitting that data as well.

So for each point that gets written, it'll probably get read at least more than a few times. Heavy for writes, even heavier for reads. I hate time series databases :)